### PR TITLE
GCU ops support RN50 training & Llama inference.

### PR DESCRIPTION
**Pull request by joeshow79**
Ops support mmpretrain resnet50 training on imagenet1k(exclude normal_)
Ops Support facebook/llama(exclude cumsum, multinomial, triu, where) inference with fairscale.

Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Checklist
**Before PR**:

- [ ] I have read and followed the workflow indicated in the [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) to create this PR.
- [ ] Pre-commit or linting tools indicated in [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) are used to fix the potential lint issues.
- [ ] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,7 +4,7 @@ __pycache__/
 *$py.class
 
 # C extensions
-*.so
+#*.so
```
[the remark this answers is not in the document]
Comment (joeshow79) [reply .gitignore:7]: Due to the underlying Enflame low-level ops currently are not about to open to community, so we need to encapsulate all the low-level symbol in libops.so and make them not visible to outside world.

File: impl/topsrider/test/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,64 @@
+set(DIOPIRT export_runtime)
+set(DIOPIFUNCTIONS export_functions)
+
+add_compile_options(-fno-elide-constructors -fpermissive)
+add_subdirectory(${CMAKE_SOURCE_DIR}/third_party/pybind11 build)
+
+include_directories(SYSTEM "${PROJECT_SOURCE_DIR}/test/include")
+include_directories(SYSTEM "${PROJECT_SOURCE_DIR}/../third_party/pybind11/include")
+include_directories(SYSTEM "${PROJECT_SOURCE_DIR}/../../dipu/third_party/DIOPI/DIOPI-IMPL/third_party/pybind11/include")
```
[the remark this answers is not in the document]
Comment (joeshow79) [reply impl/topsrider/test/CMakeLists.txt:9]: fixed

File: impl/topsrider/test/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+#include_directories(SYSTEM "/home/cse/miniconda3/envs/dipu/include/python3.8")"
+include_directories(SYSTEM "~/miniconda3/envs/dipu/include/python3.8")
```
[the remark this answers is not in the document]
Comment (joeshow79) [reply impl/topsrider/test/CMakeLists.txt:11]: Need to setup miniconda env as guide as the python version is verified in development phase, and the fixed version looks good to us so far.

File: impl/topsrider/test/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+set(RUNTIME_SRC
+    litert.cpp
+    ../conform_test.cpp)
```
[the remark this answers is not in the document]
Comment (joeshow79) [reply impl/topsrider/test/CMakeLists.txt:15]: fixed

File: impl/topsrider/test/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+target_link_libraries(${DIOPIRT} PRIVATE diopirt)
+#target_link_libraries(diopirt ${DEVICEIMPL})
+# Since all the symbols are declared weak in diopi_impl,
+# depending on the compiler and linker version,
+# linking `diopi_impl.so` may fail.
+# Here, we use no-as-needed to ensure that diopi_impl.so is linked.
+#target_link_libraries(${DIOPIRT} -Wl,--no-as-needed ${DEVICEIMPL} -Wl,--as-needed)
+target_link_libraries(diopirt -Wl,--no-as-needed ${DEVICEIMPL} -Wl,--as-needed)
+
+set(FUNCTION_SAVE_PATH "${CMAKE_SOURCE_DIR}/../DIOPI-TEST/csrc")
```
[the remark this answers is not in the document]
Comment (joeshow79) [reply impl/topsrider/test/CMakeLists.txt:36]: fixed

**Review by xintian-514 — APPROVED** [review]

**Pull request merged**